flexget/components/hook/hook.py

```diff
@@ -0,0 +1,336 @@
+from flexget.components.hook.hook_util import ENDPOINT, task_to_dict
+
+
+PLUGIN_NAME = 'hook'
+logger = logger.bind(name=PLUGIN_NAME)
+
+
+class Hook:
+    """
+    Sends a Hook
+
+    Config:
+    hook:
+      title: <<title | optional>>
+      events <<events to lisen [task|phase|plugin] optional>>
+      stages <<stages to lisen [start|end] optional>>
+      tasks  <<tasks to lisen optional>>
+      plugins <<plugins to lisen optional>>
+      when <<when to trigger [accepted|rejected|failed|no_entries|aborted] optional
+      phases <<phases to lisen>>
+      via
+        - <<hook interfaxe>>
+
+    Example:
+      hook:
+        - events:
+            - phase
+            - plugin
+          when: accepted
+          via:
+            - webhooks:
+                host: 'https://noderedurl.1880'
+                method: 'get'
+                endpoint: 'flexget/{{event_tree|join('/')}}'
+                data:
+                  task: '{{task_name}}'
+
+    """
+
+    config = {}
+
+    schema = one_or_more(
+        {
+            'type': 'object',
+            'additionalProperties': False,
+            'properties': {
+                'title': {'type': 'string'},
+                'events': one_or_more(
+                    {
+                        'type': 'string',
+                        'enum': ['task', 'phase', 'plugin'],
+                    }
+                ),
+                'stages': one_or_more({'type': 'string', 'enum': ['start', 'end']}),
+                'tasks': one_or_more({'type': 'string'}),
+                'plugins': one_or_more({'type': 'string'}),
+                'when': one_or_more(
+                    {
+                        'type': 'string',
+                        'enum': [
+                            'accepted',
+                            'rejected',
+                            'failed',
+                            'no_entries',
+                            'aborted',
+                        ],
+                    }
+                ),
+                'phases': one_or_more(
+                    {
+                        'type': 'string',
+                        'enum': plugin.task_phases,
+                    }
+                ),
+                'via': {
+                    'type': 'array',
+                    'items': {
+                        'allOf': [
+                            {'$ref': '/schema/plugins?interface=hooks'},
+                            {
+                                'maxProperties': 1,
+                                'error_maxProperties': 'Plugin options indented 2 more spaces than '
+                                'the first letter of the plugin name.',
+                                'minProperties': 1,
+                            },
+                        ]
+                    },
+                },
+            },
+        }
+    )
+
+    @staticmethod
+    def send_hook(title, data: dict, hooks, *args, **kwargs):
```

Comment:
Data is the content of the hook! The information that is send! It can be anything! Normally it's dict (that will go as JSON) but could be also a string!